### ISSUE TITLE
Default Kafka Config overrides more specific consumer/producer configuration

### ISSUE THREAD

**Issue by flansen** [smallrye/smallrye-reactive-messaging#1053]
Hi,

I'm using Smallrye Reactive Messaging with Quarkus 1.11.3.Final.

### Setup
I provide a default configuration for kafka as mentioned [in the docs](https://smallrye.io/smallrye-reactive-messaging/smallrye-reactive-messaging/2.1/kafka/kafka.html#kafka-default-configuration): 

```java
@Produces
@Named("default-kafka-broker")
public Map<String, Object> createKafkaRuntimeConfig() {
    // copy pasta from smallrye docs....
    properties.put("fetch.min.bytes", "3")
    return properties;
}
```
Now I configure my incoming kafka connector in the application.yml

```yaml
mp:
  messaging:
    incoming:
      my-incoming-stream:
        connector: smallrye-kafka
        topic: foobar
        value:
          deserializer: org.apache.kafka.common.serialization.StringDeserializer
        fetch:
          min:
            bytes: 17
```

### Expectation:
Setting `fetch.min.bytes` for `my-incoming-stream` overrides the default configuration as it is more specific.

### Actual behaviour:
The default configuration overrides the more specific consumer config.

### Probable Cause
I had a look at the code and it seems to be the implemented behaviour in `io.smallrye.reactive.messaging.kafka.KafkaConnector#merge`

```java
public <T> T getValue(String propertyName, Class<T> propertyType) {
    T t = (T) defaultKafkaCfg.get(propertyName);
    if (t == null) {
        return passedCfg.getValue(propertyName, propertyType);
    }
    return t;
}
```
The same behaviour applies to `getOptionalValue`.

As you can see, the default config takes precedence over the passed config. The passed config is only applied, when the default config does not have a value for the required propertyName.
From my understanding, this should be switched around.

**Comment by cescoffier** [comment]
You are absolutely right. It should be switched!

**Comment by flansen** [comment]
I have coded the fix, but I'm not allowed to create a new branch for this repository. If you allow me to contribute, I'll open the PR for it. :)

**Comment by cescoffier** [comment]
Thanks @flansen! 

You need to create a PR from a fork.

**Comment by flansen** [comment]
Thanks for explaining, this is my first time contributing... :)

https://github.com/smallrye/smallrye-reactive-messaging/pull/1057 
is my PR for it, I guess I can link it somehow to this issue?

**Comment by cescoffier** [comment]
Fix by #1053
